src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -996,6 +996,16 @@ def merge_kubernetes_configurations(existing_file, addition_file):
     existing = load_kubernetes_configuration(existing_file)
     addition = load_kubernetes_configuration(addition_file)
 
+    # rename the admin context so it doesn't overwrite the user context
+    try:
+        for ctx in addition['contexts']:
+            if ctx['context']['user'].startswith('clusterAdmin'):
+                admin_name = ctx['name'] + '-admin'
+                addition['current-context'] = ctx['name'] = admin_name
+                break
+    except (KeyError, TypeError):
```

Comment:
Shouldn't the try catch in the loop instead of out of the loop?

Reply:
Yes indeed.